tunnel/src/main/java/com/wireguard/config/Interface.java

```diff
@@ -366,6 +428,26 @@ public Builder parsePrivateKey(final String privateKey) throws BadConfigExceptio
             }
         }
 
+        public Builder parsePreUp(final String script) throws BadConfigException {
+            preUp.add(script);
+            return this;
+        }
+
+        public Builder parsePostUp(final String script) throws BadConfigException {
+            postUp.add(script);
+            return this;
+        }
+
+        public Builder parsePreDown(final String script) throws BadConfigException {
+            preDown.add(script);
+            return this;
+        }
+
+        public Builder parsePostDown(final String script) throws BadConfigException {
+            postDown.add(script);
+            return this;
+        }
```

Comment:
I left them because these methods are `public` and this gives the opportunity to change the parsing/validation logic without having to adjust the method signature.

I can remove them if that is preferred though